tests/optimize/test_adagrad.py

```diff
@@ -22,20 +22,24 @@ def test_apply_grad(self, grad, args, tol):
         """
         stepsize, eps = 0.1, 1e-8
         sgd_opt = AdagradOptimizer(stepsize, eps=eps)
-        grad, args = np.array(grad), np.array(args, requires_grad=True)
 
-        a1 = grad**2
-        expected = args - stepsize / np.sqrt(a1 + eps) * grad
+        grad = (np.array(grad),)
```

Comment:
Why is it put in a tuple?

Reply:
`apply_grad` accepts `grad` and `args` as tuples.  Not sure why this worked before.  Probably to do with the flattening that happened.